pkg/pyramid/tier_fs.go

```diff
@@ -256,19 +261,12 @@ func (tfs *TierFS) openFile(ctx context.Context, fileRef localFileRef, fh *os.Fi
 	}
 
 	if !tfs.eviction.Store(fileRef.fsRelativePath, stat.Size()) {
```

Comment:
OnReject is called also in case method returns false?

Reply:
Both OnEvict and OnReject CB are configured to run the same OnEvict method

Reply:
Right - but for a rejected file you will call `tfs.fileTracker.Delete(rPath)` twice. Here and in the `OnReject` callback (which calls `tfs.removeFromLocal`)

Reply:
From what I see onReject is called only in processItems (NewCache, Clear)
This flow is for Set

Reply:
Now sure I understand. `onReject` is called whenever a file isn't being stored (decided by their policy) - so here we would definitely call `Delete` twice for a rejected file.

Reply:
ristrettoEviction Store calls cache Set which in turn calls SetWithTTL.
In the SetWithTTL flow there are several places where the function return false without calling onReject CB.

Reply:
Right - `onReject` only gets called when the eviction policy decides to reject. All other rejections are not called.